src/control/server/mgmt_system.go

```diff
@@ -1138,7 +1145,59 @@ func (svc *mgmtSvc) SystemSetProp(ctx context.Context, req *mgmtpb.SystemSetProp
 		return nil, err
 	}
 
-	return &mgmtpb.DaosResp{}, nil
+	// for any system property that has a pool property, update all pools
```

Comment:
This is a pretty large increase to the `SystemSetProp()` RPC handler with logic that is ancillary to its primary purpose. It would be better to break it out into a helper function that can be unit tested in isolation, I think.

It's not a rule, exactly, more of a goal, but in general we are trying to keep these RPC handlers relatively thin in order to encourage better separation of concerns and easier unit testing.

Reply:
Refactored ... let me know what you think.